gcloud-java-dns/README.md

```diff
@@ -0,0 +1,324 @@
+Cloud DNS for your project.
+
+See the [``gcloud-java-dns`` API documentation][dns-api] to learn how to interact
+with the Cloud DNS using this client Library.
+
+Getting Started
+---------------
+#### Prerequisites
+For this tutorial, you will need a [Google Developers Console](https://console.developers.google.com/)
+project with the DNS JSON API enabled. You will need to [enable billing](https://support.google.com/cloud/answer/6158867?hl=en)
+to use Google Cloud DNS. [Follow these instructions](https://cloud.google.com/docs/authentication#preparation)
+to get your project set up. You will also need to set up the local development environment by
+[installing the Google Cloud SDK](https://cloud.google.com/sdk/) and running the following commands
+in command line: `gcloud auth login` and `gcloud config set project [YOUR PROJECT ID]`.
+
+#### Installation and setup
+You'll need to obtain the `gcloud-java-dns` library.  See the [Quickstart](#quickstart) section to
+add `gcloud-java-dns` as a dependency in your code.
+
+#### Creating an authorized service object
+To make authenticated requests to Google Cloud DNS, you must create a service object with
+credentials. You can then make API calls by calling methods on the DNS service object. The simplest
+way to authenticate is to use [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials).
+These credentials are automatically inferred from your environment, so you only need the following
+code to create your service object:
+
+```java
+import com.google.gcloud.dns.Dns;
+import com.google.gcloud.dns.DnsOptions;
+
+Dns dns = DnsOptions.defaultInstance().service();
+```
+
+For other authentication options, see the [Authentication](https://github.com/GoogleCloudPlatform/gcloud-java#authentication) page.
+
+#### Managing Zones
+DNS records in `gcloud-java-dns` are managed inside containers called "zones". `ZoneInfo` is a class
+which encapsulates metadata that describe a zone in Google Cloud DNS, but it cannot interact with
+the Cloud DNS service on its own. `Zone`, a subclass of `ZoneInfo`, adds service-related
+functionality over `ZoneInfo`. In this code snippet, we create a new zone in which we intend to
+manage DNS record for domain `someexampledomain.com.`
+
+Add the following imports at the top of your file:
+
+```java
+import com.google.gcloud.storage.Zone;
+import com.google.gcloud.storage.ZoneInfo;
+```
+
+Then add the following code to create a zone.
+
+*Important: Zone names within each project have to be unique. If you choose a zone name that already
+exists, you'll get a helpful error message telling you to choose another name. In the code below,
+replace "my_unique_zone" with a unique zone name. See more about naming rules [here](https://cloud.google.com/dns/api/v1/managedZones).*
+
+*Important: The service may require that you verify ownership of the domain name that you are
```

Comment:
grammar nit: "The service may require that you verify ownership of the domain for which you are creating a zone".